### PR TITLE
Add ability to share conversation transcript with link

### PR DESCRIPTION
**Pull request by reichert621**
### Description

Adds the basic ability to share a conversation transcript with a link that expires after 24 hours.

### Issue

Sets up a v1 of https://github.com/papercups-io/papercups/issues/457

### Screenshots

share-conversation-v1

## Checklist

- [x] Everything passes when running `mix test`
- [x] Ran `mix format`
- [x] No frontend compilation warnings

### CODE REVIEW

File: lib/chat_api_web/controllers/conversation_controller.ex [hunk 1]
```diff
@@ -75,6 +75,37 @@ defmodule ChatApiWeb.ConversationController do
     render(conn, "index.json", conversations: conversations)
   end
 
+  @spec share(Plug.Conn.t(), map()) :: Plug.Conn.t()
```
Comment (reichert621) [inline lib/chat_api_web/controllers/conversation_controller.ex:78]: (still need to add tests for these endpoints)

File: lib/chat_api_web/controllers/conversation_controller.ex [hunk 2]
```diff
@@ -75,6 +75,37 @@ defmodule ChatApiWeb.ConversationController do
...
+         %{customer_id: customer_id} <- Conversations.get_conversation!(conversation_id) do
+      token = Phoenix.Token.sign(ChatApiWeb.Endpoint, conversation_id, {account_id, customer_id})
+
+      json(conn, %{data: %{ok: true, token: token}})
+    end
+  end
+
+  @spec shared(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def shared(conn, %{"conversation_id" => conversation_id, "token" => token}) do
+    case Phoenix.Token.verify(ChatApiWeb.Endpoint, conversation_id, token, max_age: 86400) do
```
Comment (reichert621) [inline lib/chat_api_web/controllers/conversation_controller.ex:90]: (maybe make it clearer that `86400` is 24 hours)

**Comment by cheeseblubber** [comment]
Would it have been easier if we have a route for the conversation id and then we wouldn't need the token and just rely on our auth? This way if the person receiving the link can navigate to the conversation to reply. Because right now if a user that received the link wants to reply to that conversation this still wouldn't solve that issue

**Comment by reichert621** [comment]
ah i guess i misunderstood the ticket 😑

**Comment by fuziontech** [comment]
This is perfect. Thanks!

**Pull request merged**